### ISSUE TITLE
[BUG] no suitable codecs found

### ISSUE THREAD

**Issue by migrau** [nf-core/sarek#367]
## Description of the bug

When trying to run sarek (Mutect2) in a cluster, it stops after this error:

```
-[nf-core/sarek] Pipeline completed with errors-
Error executing process > 'Mutect2 (lv-2_vs_lv-4-11_1-122082543)'

Caused by:
  Process `Mutect2 (lv-2_vs_lv-4-11_1-122082543)` terminated with an error exit status (2)

Command executed:

  # Get raw calls
  gatk --java-options "-Xmx32g"       Mutect2       -R genome.fa      -I lv-2.recal.bam  -tumor lv-2       -I lv-4.recal.bam -normal lv-4       -L 11_1-122082543.bed              --germline-resource input.9              -O 11_1-122082543_lv-2_vs_lv-4.vcf

Command exit status:
  2

Command output:
  (empty)

Command error:
  Using GATK jar /opt/conda/envs/nf-core-sarek-2.7/share/gatk4-4.1.7.0-0/gatk-package-4.1.7.0-local.jar
  Running:
      java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx32g -jar /opt/conda/envs/nf-core-sarek-2.7/share/gatk4-4.1.7.0-0/gatk-package-4.1.7.0-local.jar Mutect2 -R genome.fa -I lv-2.recal.bam -tumor lv-2 -I lv-4.recal.bam -normal lv-4 -L 11_1-122082543.bed --germline-resource input.9 -O 11_1-122082543_lv-2_vs_lv-4.vcf
  11:26:11.367 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/opt/conda/envs/nf-core-sarek-2.7/share/gatk4-4.1.7.0-0/gatk-package-4.1.7.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
  Apr 19, 2021 11:26:11 AM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine
  INFO: Failed to detect whether we are running on Google Compute Engine.
  11:26:11.922 INFO  Mutect2 - ------------------------------------------------------------
  11:26:11.922 INFO  Mutect2 - The Genome Analysis Toolkit (GATK) v4.1.7.0
  11:26:11.922 INFO  Mutect2 - For support and documentation go to https://software.broadinstitute.org/gatk/
  11:26:11.922 INFO  Mutect2 - Executing as mgrau@hh-yoda-01-07.ebi.ac.uk on Linux v3.10.0-693.5.2.el7.x86_64 amd64
  11:26:11.922 INFO  Mutect2 - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_265-b11
  11:26:11.922 INFO  Mutect2 - Start Date/Time: April 19, 2021 11:26:11 AM UTC
  11:26:11.923 INFO  Mutect2 - ------------------------------------------------------------
  11:26:11.923 INFO  Mutect2 - ------------------------------------------------------------
  11:26:11.923 INFO  Mutect2 - HTSJDK Version: 2.21.2
  11:26:11.923 INFO  Mutect2 - Picard Version: 2.21.9
  11:26:11.923 INFO  Mutect2 - HTSJDK Defaults.COMPRESSION_LEVEL : 2
  11:26:11.923 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
  11:26:11.923 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
  11:26:11.923 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
  11:26:11.923 INFO  Mutect2 - Deflater: IntelDeflater
  11:26:11.923 INFO  Mutect2 - Inflater: IntelInflater
  11:26:11.923 INFO  Mutect2 - GCS max retries/reopens: 20
  11:26:11.923 INFO  Mutect2 - Requester pays: disabled
  11:26:11.924 INFO  Mutect2 - Initializing engine
  11:26:12.544 INFO  Mutect2 - Shutting down engine
  [April 19, 2021 11:26:12 AM UTC] org.broadinstitute.hellbender.tools.walkers.mutect.Mutect2 done. Elapsed time: 0.02 minutes.
  Runtime.totalMemory()=2359820288
  ***********************************************************************
  
  A USER ERROR has occurred: Cannot read file://input.9 because no suitable codecs found
  
  ***********************************************************************
  Set the system property GATK_STACKTRACE_ON_USER_EXCEPTION (--java-options '-DGATK_STACKTRACE_ON_USER_EXCEPTION=true') to print the stack trace.
```

On the other hand, Strelka2 finishes without errors.

## Steps to reproduce

1. Command line:
nextflow run nf-core/sarek -profile singularity --genome GRCm38 --input input.tsv --tools 'Mutect2,VEP' --resume

## System

- Hardware: HPC 
- Executor: LSF

## Nextflow Installation

- Version: 20.10.0 build 5430
- 
## Container engine

- Engine: Singularity  
- version: 3.5.0

**Comment by MaxUlysse** [comment]
Hi @migrau
There was an error with the germline resource file.
If you check the actual command line that was run:
```
gatk --java-options "-Xmx32g"       Mutect2       -R genome.fa      -I lv-2.recal.bam  -tumor lv-2       -I lv-4.recal.bam -normal lv-4       -L 11_1-122082543.bed              --germline-resource input.9              -O 11_1-122082543_lv-2_vs_lv-4.vcf
```
You can see that this parameter is not properly defined `--germline-resource input.9`.
Which is easily explained by the fact that we don't have this file for mice.
I'm afraid GATK is not providing any non-human files.
We're currently working on the next Sarek release which will be a DSL2 one, and we'll take into account the possibility of running Mutect2 without this file (which is not necessary, but just recommended with the Panel-of-Normals).
If you really want to run Sarek with some tiny modifications, I can guide you into changing that in your local copy of sarek.
(basically, just removing this line from the script: https://github.com/nf-core/sarek/blob/master/main.nf#L2445).

**Comment by migrau** [comment]
Hi @maxulysse 
I removed the line (L2445) from the main script, but I still get an error (now from the GetPileupSummaries function). I removed the L2587 (`-V ${germlineResource}`) but actually the argument 'variant' is required, so it doesn't work. Thanks!

**Comment by maxulysse** [comment]
Hi @migrau 
I see, then it is needed.
I'm afraid I'm not sure on how to process with that.

**Comment by maxulysse** [comment]
related to #314

**Comment by dataMaster-Kris** [comment]
I get a similar error for mouse data. So I launched the singularity shell from Sarek's `work/singularity` folder and tested the Mutect2 command without the germline resource option and it ran fine. Example:

`java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx7g -jar /opt/conda/envs/nf-core-sarek-2.7.1/share/gatk4-4.1.7.0-0/gatk-package-4.1.7.0-local.jar Mutect2 -R genome.fa  -I T8_S1.recal.bam  -tumor T8_S1 -L 1_1-195471971.bed -O 1_1-195471971_T8_S1.vcf`

The [Mutect2 website](https://gatk.broadinstitute.org/hc/en-us/articles/360042477952-Mutect2) states that the germline resource is not required. What is my best option? Should I manually run the above command using the sarek singularity container, and later resume the sarek pipeline from the next steps? Would resuming work? It would be great if there were a simple way to turn off the germline resource requirement imposed by sarek. Thanks.

**Comment by maxulysse** [comment]
For now, the best option would be to run manually the Mutect2 part.
This germline resource requirement will be handled properly in the next release

**Comment by samkleeman1** [comment]
Any updates on this? I am having the same issue.

**Comment by mheskett** [comment]
Just adding my vote. I am running into the same exact issue. Any update? Will the pipeline work with one of the other mutation callers?

**Comment by FriederikeHanssen** [comment]
yes other variantvallers would work and we can make the germline resource file optional in the new release

**Comment by FriederikeHanssen** [comment]
Closed by #592
